src/virtualenv/seed/wheels/periodic_update.py

```diff
@@ -291,6 +318,8 @@ def _run_do_update(app_data, distribution, embed_filename, for_py_version, perio
     if not u_log.periodic:
         u_log.started = now
     u_log.versions = versions + u_log.versions
+    # we need to sort versions for the next update to work properly
+    u_log.versions.sort(reverse=True)
```

Comment:
The log needs to be sorted. It was sorted by design before. That's not the case anymore.

Reply:
Can you explain why is no longer the case?

Reply:
> This would remove the need to sort the list. 

I don't mind sorting the log; why would you consider the other solution less risky?

Reply:
> Can you explain why is no longer the case?

because other sources logs are always appended rather than inserted.

> I don't mind sorting the log; why would you consider the other solution less risky?

I'm feeling a bit hesitant to rely on the tuple ordering given:
```
    def as_version_tuple(version):
        result = []
        for part in version.split(".")[0:3]:
            try:
                result.append(int(part))
            except ValueError:
                break
        if not result:
            raise ValueError(version)
        return tuple(result)
```
I feel that the inner try/catch is hiding something I did not think about.

Reply:
I did not think of pre-release versions... In this specific sample, it does not matter much but it's not sorted correctly.
```
{
  "completed": "2022-01-01T21:48:50.557509Z",
  "periodic": false,
  "started": "2022-01-01T21:48:40.230835Z",
  "versions": [
    {
      "filename": "pip-21.3.1-py3-none-any.whl",
      "found_date": "2022-01-01T14:22:37.717326Z",
      "release_date": "2021-10-22T15:57:09.000000Z",
      "source": "manual"
    },
    {
      "filename": "pip-20.3-py2.py3-none-any.whl",
      "found_date": "2022-01-01T21:46:25.286240Z",
      "release_date": "2020-11-30T12:47:49.000000Z",
      "source": "manual"
    },
    {
      "filename": "pip-20.0-py2.py3-none-any.whl",
      "found_date": "2022-01-01T21:48:23.634459Z",
      "release_date": "2020-01-21T11:42:56.000000Z",
      "source": "manual"
    },
    {
      "filename": "pip-20.3b1-py2.py3-none-any.whl",
      "found_date": "2022-01-01T21:44:30.798536Z",
      "release_date": "2020-10-31T19:25:12.000000Z",
      "source": "manual"
    }
  ]
}
```
Further-more, a manual/periodic update should never lead to defaulting to pre-release version. This promise is broken (at least for manual updates).

Reply:
Given an added test with a pre-release version downloaded as "pinned" version indeed broke manual updates, I went with the alternate implementation of _run_do_update which first splits the UpdateLog in 2 depending if it's coming from a previous update or another source.